### ISSUE TITLE
Softmax + concat does not work with cuDNN enabled ?

### ISSUE THREAD

**Issue by DianeBouchacourt** [chainer/chainer#1605]
I have implemented a simple neural network and a true multi-class Hinge Loss to perform multiclass classification. However the behavior of the backpropagation seems to differ when cuDNN is enabled or not in the softmax function, due to the use of F.concat. I have attached 2 python files, run "python small_example.py" and change use_cnn to False/True to observe the different behavior (accuracy 95 % after 10 epochs if disabled, 22% if enabled). Maybe it's a bug on my cuDNN.

[issue_chainer.zip](https://github.com/pfnet/chainer/files/454900/issue_chainer.zip)

**Comment by delta2323** [comment]
Hi, thank you for reporting the problem. I am afraid it is hard for me to identify where the problem is. So, is it possible for you to make a minimal code in which problem occurs so that we can narrow the problem?

**Comment by DianeBouchacourt** [comment]
I have enclosed a simple example code, you mean something even more minimal
?

**Comment by delta2323** [comment]
> you mean something even more minimal?

Yes, it is what I thought in mind. As it involves some work, it is OK as much as you can. But I would like you to extract the point where some inconsistencies occur as much as possible.
